core/src/test/java/org/apache/gravitino/storage/TestSQLScripts.java

```diff
@@ -0,0 +1,166 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class TestSQLScripts extends BaseIT {
+
+  private String jdbcBackend;
+  private Path scriptDir;
+  @TempDir private File tempDir;
+
+  @BeforeAll
+  public void startIntegrationTest() {
+    jdbcBackend = System.getenv("jdbcBackend");
+    Assertions.assertNotNull(jdbcBackend, "jdbcBackend environment variable is not set");
+
+    String gravitinoHome = System.getenv("GRAVITINO_HOME");
+    Assertions.assertNotNull(gravitinoHome, "GRAVITINO_HOME environment variable is not set");
+    scriptDir = Path.of(gravitinoHome, "scripts", jdbcBackend.toLowerCase());
+    Assertions.assertTrue(Files.exists(scriptDir), "Script directory does not exist: " + scriptDir);
+  }
+
+  @Test
+  public void testSQLScripts() throws IOException, SQLException {
+    File[] scriptFiles = scriptDir.toFile().listFiles();
+    Assertions.assertNotNull(scriptFiles, "No script files found in " + scriptDir);
+    // Sort files to ensure correct execution order (schema -> upgrade)
+    Arrays.sort(scriptFiles, Comparator.comparing(File::getName));
+
+    // A map to store connections for different schema versions
+    Map<String, Connection> versionConnections = new HashMap<>();
+    Pattern schemaPattern =
+        Pattern.compile("schema-([\\d.]+)-" + jdbcBackend.toLowerCase() + "\\.sql");
+    Pattern upgradePattern =
+        Pattern.compile("upgrade-([\\d.]+)-to-([\\d.]+)-" + jdbcBackend.toLowerCase() + "\\.sql");
+
+    for (File scriptFile : scriptFiles) {
+      Matcher schemaMatcher = schemaPattern.matcher(scriptFile.getName());
+      Matcher upgradeMatcher = upgradePattern.matcher(scriptFile.getName());
+
+      if (schemaMatcher.matches()) {
+        String version = schemaMatcher.group(1);
+        String dbName = "schema_" + version.replace('.', '_');
+
+        Connection conn = getSQLConnection(jdbcBackend, dbName);
+        Assertions.assertDoesNotThrow(
+            () -> executeSQLScript(conn, scriptFile),
+            "Failed to execute schema script for version " + version);
+        versionConnections.put(version, conn);
+
+      } else if (upgradeMatcher.matches()) {
+        String fromVersion = upgradeMatcher.group(1);
+
+        Connection conn = versionConnections.get(fromVersion);
+        Assertions.assertNotNull(
+            conn, "No existing database connection found for version " + fromVersion);
+        Assertions.assertDoesNotThrow(
+            () -> executeSQLScript(conn, scriptFile),
+            "Failed to execute upgrade script" + " in file " + scriptFile.getName());
+      } else {
+        Assertions.fail("Unrecognized script file name: " + scriptFile.getName());
+      }
+    }
+
+    // Close all connections
+    for (Connection conn : versionConnections.values()) {
+      if (conn != null) {
+        conn.close();
+      }
+    }
+  }
```

Comment:
I would suggest closing all the connections in the `AfterAll` method. Also, do you need to clean the database after the test is finished?